### PR TITLE
Add option for custom migrations table (fixes #4)

### LINKED ISSUE #4: Pass in custom migrations table name
Component doesn't allow to pass in custom migrations table name.

### PR DESCRIPTION
**Pull request by mariusz-jachimowicz-83**
Add option for custom migrations table (fixes #4)

### CODE REVIEW

File: src/duct/migrator/ragtime.clj [hunk 1]
```diff
@@ -59,8 +59,13 @@
   (update migration :id str "#" (subs (hash-migration migration) 0 8)))
 
 (defn- migrate [index {:keys [database migrations strategy logger]
-                       :or   {strategy :raise-error}}]
-  (ragtime/migrate-all (jdbc/sql-database (:spec database)) index migrations
+                       :or   {strategy :raise-error}
+                       :as   opts}]
+  (ragtime/migrate-all (jdbc/sql-database (:spec database)
+                                          (or (select-keys opts [:migrations-table])
```
Comment (weavejester) [inline src/duct/migrator/ragtime.clj]: Can you put this in the argument destructuring instead?

File: test/duct/migrator/ragtime_test.clj [hunk 2]
```diff
@@ -45,21 +45,38 @@
...
-    (is (= (find-tables)
-           [{:name "ragtime_migrations"}
-            {:name "foo"}
-            {:name "bar"}]))
-    (is (= @logs
-           [[::ragtime/applying ":duct.migrator.ragtime-test/create-foo#f1480e44"]
-            [::ragtime/applying ":duct.migrator.ragtime-test/create-bar#6d969ce8"]]))))
+  (drop-all-tables))
+
+(use-fixtures :each (fn [test-fn]
```
Comment (weavejester) [inline test/duct/migrator/ragtime_test.clj]: Can you avoid use of `use-fixtures`?

**Review by weavejester — CHANGES_REQUESTED** [review]

**Comment by weavejester** [comment]
Thanks for the PR. I've marked a couple of changes. Can you also ensure that your commit message is consistent with the repository, so something more like:

    Add option for custom migrations table (fixes #4)

File: test/duct/migrator/ragtime_test.clj (hunk 2 above)
Comment (mariusz-jachimowicz-83) [reply test/duct/migrator/ragtime_test.clj]: Sure I will apply it. Why do you prefer to avoid them?

File: src/duct/migrator/ragtime.clj (hunk 1 above)
Comment (mariusz-jachimowicz-83) [reply src/duct/migrator/ragtime.clj]: Sure I will apply it.

**Comment by mariusz-jachimowicz-83** [comment]
@weavejester code review tips applied. If all is ok I will squash commits.

File: src/duct/migrator/ragtime.clj [hunk 3]
```diff
@@ -58,9 +58,14 @@
 (defn- add-hash-to-id [migration]
   (update migration :id str "#" (subs (hash-migration migration) 0 8)))
 
-(defn- migrate [index {:keys [database migrations strategy logger]
+(defn- migrate [index {:keys [database migrations strategy logger migrations-table]
                        :or   {strategy :raise-error}}]
-  (ragtime/migrate-all (jdbc/sql-database (:spec database)) index migrations
+  (ragtime/migrate-all (jdbc/sql-database (:spec database)
+                                          (if migrations-table
```
Comment (weavejester) [inline src/duct/migrator/ragtime.clj]: That's looking a little hard to understand. Perhaps we write something more like:

```clojure
(defn- get-database [{{:keys [spec]} :database :as opts}]
  (jdbc/sql-database spec (select-keys opts [:migrations-table])))

(defn- get-strategy [{:keys [strategy] :or {strategy :raise-error}]
  (strategies strategy))

(defn- get-reporter [{:keys [logger]}]
  (if logger (logger-reporter logger) reporter/print))

(defn- migrate [index {:keys [migrations] :as opts}]
  (let [db    (get-database opts)
        strat (get-strategy opts)
        rep   (get-reporter opts)]
    (ragtime/migrate-all db index migrations {:reporter rep, :strategy strat})
    (ragtime/into-index index migrations)))
```

**Review by weavejester — CHANGES_REQUESTED** [review]

File: src/duct/migrator/ragtime.clj [hunk 4]
```diff
@@ -58,14 +58,21 @@
...
+
+(defn- get-reporter [{:keys [logger]}]
+  (if logger (logger-reporter logger) reporter/print))
+
+(defn- migrate [index {:keys [migrations] :as opts}]
+  (let [db    (get-database opts)
+        strat (get-strategy opts)
+        rep   (get-reporter opts)]
+    (ragtime/migrate-all db index migrations {:reporter rep, :strategy strat})
+    (ragtime/into-index index migrations)))
```
Comment (mariusz-jachimowicz-83) [inline src/duct/migrator/ragtime.clj:75]: @weavejester refactored as requested,

**Comment by mariusz-jachimowicz-83** [comment]
Yo @weavejester could you make some decision about this PR ? It's very disappointing when PR is hanging a long time.

**Comment by weavejester** [comment]
Sometimes pull requests get pushed down my priority list by more urgent matters, and older PRs can get forgotten about as new ones pile up. If a PR is taking a long time, prod me about it to remind me.

I manage around 60 open source projects, along with the work I'm actually paid for, so while it might be disappointing that I sometimes need reminding that PRs exist, please bear in mind that I'm doing this for free.

**Comment by mariusz-jachimowicz-83** [comment]
@weavejester Aaa big number. :smiley: Hope that you will have a lot more of time to push your ideas much further. I admire you current work :smiley:

**Pull request merged**